### PR TITLE
fix(networkport): fix behavior on delete

### PR DESCRIPTION
**Pull request by stonebuzz**
Actually, when a ```NetworkPort``` is deleted, 

From ```dynamic``` asset, ```NetworkPort``` is always displayed from ```Network``` tab, while it is already present in the ```LockNetwork``` tab.

This PR exclude ```deleted``` ```NetworkPort``` when main item is ```dynamic``` (purge can only be done from ```Lock``` tab or from ```NetworkPort```  form.

From ```manual``` asset, ```NetworkPort``` is always displayed from ```Network``` tab, while it is already present in the ```LockNetwork``` tab.
This PR add ```Deleted``` column to ```NetowrkPort``` list (usefull for manual asset)

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !25618

### CODE REVIEW

File: install/migrations/update_10.0.5_to_10.0.6/networkport.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ * @var array $ADDTODISPLAYPREF
+ */
+$ADDTODISPLAYPREF['NetworkPort'] = [3, 30, 31, 32, 33, 34, 35, 36, 38, 39, 40, 6];
```
Comment (cedric-anne) [inline install/migrations/update_10.0.5_to_10.0.6/networkport.php:41]:
```suggestion
$ADDTODISPLAYPREF['NetworkPort'] = [6];
```

**Review by cedric-anne — APPROVED** [review]

File: install/migrations/update_10.0.5_to_10.0.6/networkport.php (hunk 1 above)
Comment (stonebuzz) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php:41]: This way add new entry to ```glpi_displaypreferences``` but with ```rank``` set to ```1```

Comment (stonebuzz) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php:41]: Before : 

```sql
MariaDB [10bugfixes_ruless]> select * from glpi_displaypreferences where itemtype = "NetworkPort";
+-----+-------------+-----+------+----------+
| id  | itemtype    | num | rank | users_id |
+-----+-------------+-----+------+----------+
| 266 | NetworkPort |   3 |    1 |        0 |
| 267 | NetworkPort |  30 |    2 |        0 |
| 268 | NetworkPort |  31 |    3 |        0 |
| 269 | NetworkPort |  32 |    4 |        0 |
| 270 | NetworkPort |  33 |    5 |        0 |
| 271 | NetworkPort |  34 |    6 |        0 |
| 272 | NetworkPort |  35 |    7 |        0 |
| 273 | NetworkPort |  36 |    8 |        0 |
| 274 | NetworkPort |  38 |    9 |        0 |
| 275 | NetworkPort |  39 |   10 |        0 |
| 276 | NetworkPort |  40 |   11 |        0 |
+-----+-------------+-----+------+----------+
11 rows in set (0.000 sec)
```

After 

```sql
MariaDB [10bugfixes_ruless]> select * from glpi_displaypreferences where itemtype = "NetworkPort";
+-----+-------------+-----+------+----------+
| id  | itemtype    | num | rank | users_id |
+-----+-------------+-----+------+----------+
| 266 | NetworkPort |   3 |    1 |        0 |
| 267 | NetworkPort |  30 |    2 |        0 |
| 268 | NetworkPort |  31 |    3 |        0 |
| 269 | NetworkPort |  32 |    4 |        0 |
| 270 | NetworkPort |  33 |    5 |        0 |
| 271 | NetworkPort |  34 |    6 |        0 |
| 272 | NetworkPort |  35 |    7 |        0 |
| 273 | NetworkPort |  36 |    8 |        0 |
| 274 | NetworkPort |  38 |    9 |        0 |
| 275 | NetworkPort |  39 |   10 |        0 |
| 276 | NetworkPort |  40 |   11 |        0 |
| 310 | NetworkPort |   6 |    1 |        0 |
+-----+-------------+-----+------+----------+
12 rows in set (0.001 sec)
```

Comment (cedric-anne) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php:41]: I already noticed this, indeed. We should probably handle it in migration class, globally, and in another PR.

Comment (stonebuzz) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php]: See [#0fc08ad](https://github.com/glpi-project/glpi/pull/13349/commits/0fc08ad82235d927a14e863e0c15380d5fbb4fa5), which used ```updateDisplayPrefs``` from ```MLigration``` class

Comment (trasher) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php]: :+1:

**Review by cedric-anne — APPROVED** [review]

File: install/migrations/update_10.0.5_to_10.0.6/networkport.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ * @var array $ADDTODISPLAYPREF
+ */
+ // Add display preferences
+$migration->updateDisplayPrefs(
```
Comment (cedric-anne) [inline install/migrations/update_10.0.5_to_10.0.6/networkport.php:42]: `updateDisplayPrefs` seems to add the display pref on users personnal preferences also. Could you add a new `bool $only_default = false` parameter that would be used to update only default preferences (users_id  0) when set to `true` ?

Comment (stonebuzz) [reply install/migrations/update_10.0.5_to_10.0.6/networkport.php:42]: done with https://github.com/glpi-project/glpi/pull/13349/commits/0127aed26e7587b825bd8b9f790db5e3aa68e115

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Comment by cedric-anne** [comment]
I pushed a commit to refactor a little the `Migration::updateDisplayPrefs()` method. Also, I used it globally.

**Pull request merged**